### PR TITLE
Sign up form xstate integ test

### LINKED ISSUE #482: SignUp form and logic
First step ( without email verification ):
client:
- [x] Create a `signUp` screen in the existing `AuthenticationStack`, inside the screen create a form containing an email input username input and password input but also a submit button, parse the email and password on the fly, user should not be able to send a request if email is correctly formated `address@domain.extension` and if the password is not **strong** enough
- [x] On request failure depending on error status ( `username` or `email` ) display specific form error
- [x] On request success the user is authenticated and he should be automatically redirected to the application ~~or ?? user is redirected to the `signIn` screen??~~
- [x] tests

server:
- [x] create a `SignUp` route that takes `userName, email, password, authenticationMode`
- [x] Handle SignUp route, verify `username` and `email` availability to send back custom errors, but also check email is valid and password is **strong**
- [x] After parsing request body, create the user in database, sign in the user depending on the given `authenticationMode` and sendback the authenticated response ~~or ?? send back a 200 response ??~~
- [x] tests

### PR DESCRIPTION
**Pull request by prastoin**
close #482

### CODE REVIEW

File: packages/client/__tests__/Authentication.test.tsx [hunk 1]
```diff
@@ -534,6 +1087,61 @@ const authenticationModelMachine =
...
+                                equals: signUpEmail,
+                            },
+                        },
+                    });
+
+                    if (userWithSameEmail) {
+                        return true;
+                    }
+                    return false;
+                },
```
Comment (Devessier) [inline packages/client/__tests__/Authentication.test.tsx:1122]: 😳😍

**Comment by prastoin** [comment]
Updated the PR to be redirecting on home avec successfull sign up + tests
Also testing unkown server error management in the client in tests

**Comment by prastoin** [comment]
I still need to add a or sign in hyperlink inside sign up form screen and test it !

**Review by Devessier — APPROVED** [review]
Absolutely amazing work! This is the beginning of an accurate use of `@xstate/test` for us.
We can merge.

**Pull request merged**